File: dev/python/tests/test_schema.py
import shutil

from drugbank_parse.schema import default_schema_dir, load_schema


def test_load_schema_returns_core_tables():
    schema = load_schema()

    assert schema.version == 1
    assert list(schema.tables) == [
        "drugs",
        "targets",
        "drug_target",
        "drug_indication",
        "target_drug_indication",
    ]
    assert schema.tables["drugs"].columns == [
        "drug_id",
        "drug_name",
        "inchi",
        "source",
    ]


def test_default_schema_dir_uses_packaged_schema_data():
    schema_dir = default_schema_dir()

    assert schema_dir.name == "schema_data"
    assert (schema_dir / "tables.yml").exists()
    assert (schema_dir / "fields.yml").exists()
    assert (schema_dir / "profiles.yml").exists()


def test_load_schema_accepts_schema_dir_override(project_root, tmp_path):
    schema_dir = tmp_path / "schema"
    shutil.copytree(project_root / "dev" / "schema", schema_dir)

    schema = load_schema(schema_dir=schema_dir)

    assert schema.version == 1
    assert "drugs" in schema.tables

File: dev/R/tests/testthat/test-schema.R
test_that("load_drugbank_schema reads shared core tables", {
  schema <- load_drugbank_schema(shared_schema_dir())

  expect_equal(schema$version, 1)
  expect_equal(names(schema$tables), c(
    "drugs",
    "targets",
    "drug_target",
    "drug_indication",
    "target_drug_indication"
  ))
  expect_equal(schema$tables$drugs$columns, c("drug_id", "drug_name", "inchi", "source"))
})

test_that("resolve_modules reads core profile", {
  expect_equal(resolve_modules(profile = "core", modules = NULL, schema_dir = shared_schema_dir()), "core")
  expect_equal(resolve_modules(profile = "core", modules = c("core"), schema_dir = shared_schema_dir()), "core")
})

test_that("resolve_modules rejects unknown profile and module", {
  expect_error(resolve_modules(profile = "missing", schema_dir = shared_schema_dir()), "Unknown profile")
  expect_error(resolve_modules(profile = "core", modules = c("missing"), schema_dir = shared_schema_dir()), "Unknown module")
})

test_that("resolve_tables preserves core table order", {
  expect_equal(resolve_tables(c("core"), schema_dir = shared_schema_dir()), c(
    "drugs",
    "targets",
    "drug_target",
    "drug_indication",
    "target_drug_indication"
  ))
})
